cluster/store/retry_schema.go

```diff
@@ -0,0 +1,152 @@
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2024 Weaviate B.V. All rights reserved.
+//
+//  CONTACT: hello@weaviate.io
+//
+
+package store
+
+import (
+	"github.com/cenkalti/backoff/v4"
+	"github.com/weaviate/weaviate/cluster/utils"
+	"github.com/weaviate/weaviate/entities/models"
+	"github.com/weaviate/weaviate/usecases/sharding"
+)
+
+// retrySchema is used for retrying schema queries. It is a thin wrapper around
+// the original schema, separating retry logic from the actual operation.
+// Retry may be needed due to eventual consistency issues where
+// updates might take some time to arrive at the follower.
+type retrySchema struct {
+	schema *schema
+}
+
+func (rs retrySchema) ClassInfo(class string) (ci ClassInfo) {
+	rs.retry(func(s *schema) error {
+		ci = rs.schema.ClassInfo(class)
+		if !ci.Exists {
+			return errClassExists
+		}
+		return nil
+	})
+	return
+}
+
+// ClassEqual returns the name of an existing class with a similar name, and "" otherwise
+// strings.EqualFold is used to compare classes
+func (rs retrySchema) ClassEqual(name string) string {
+	return rs.schema.ClassEqual(name)
+}
+
+func (rs retrySchema) MultiTenancy(class string) models.MultiTenancyConfig {
+	return rs.metaClass(class).MultiTenancyConfig()
+}
+
+// Read performs a read operation `reader` on the specified class and sharding state
+func (rs retrySchema) Read(class string, reader func(*models.Class, *sharding.State) error) error {
+	return rs.retry(func(s *schema) error {
+		return rs.schema.Read(class, reader)
+	})
+}
+
+// ShardOwner returns the node owner of the specified shard
+func (rs retrySchema) ReadOnlyClass(class string) (cls *models.Class) {
+	rs.retry(func(s *schema) error {
+		if cls = s.ReadOnlyClass(class); cls == nil {
+			return errClassNotFound
+		}
+		return nil
+	})
+	return
+}
+
+func (rs retrySchema) metaClass(class string) (meta *metaClass) {
+	rs.retry(func(s *schema) error {
+		if meta = s.metaClass(class); meta == nil {
+			return errClassNotFound
+		}
+		return nil
+	})
+	return
+}
+
+// ReadOnlySchema returns a read only schema
+// Changing the schema outside this package might lead to undefined behavior.
+//
+// it creates a shallow copy of existing classes
+//
+// This function assumes that class attributes are being overwritten.
+// The properties attribute is the only one that might vary in size;
+// therefore, we perform a shallow copy of the existing properties.
+// This implementation assumes that individual properties are overwritten rather than partially updated
+func (rs retrySchema) ReadOnlySchema() models.Schema {
+	return rs.schema.ReadOnlySchema()
+}
+
+// ShardOwner returns the node owner of the specified shard
+func (rs retrySchema) ShardOwner(class, shard string) (owner string, err error) {
+	err = rs.retry(func(s *schema) error {
+		owner, err = rs.schema.ShardOwner(class, shard)
```

Comment:
**suggestion (non-blocking, blocking):** similar to above  `rs` or `s`

Reply:
there is multiple functions in the file has the same style

Reply:
Good Catch!